src/main/host/network_interface.c

```diff
@@ -452,8 +452,7 @@ void networkinterface_receivePackets(NetworkInterface* interface, Host* host) {
             break;
         }
 
-        guint64 length = (guint64)(packet_getPayloadLength(packet) + packet_getHeaderSize(packet) +
-                                   CONFIG_HEADER_SIZE_ETH);
+        guint64 length = (guint64)(packet_getPayloadLength(packet) + packet_getHeaderSize(packet));
```

Comment:
In a couple places throughout the code, you're using the snippet `packet_getPayloadLength(packet) + packet_getHeaderSize(packet)`.

You could de-duplicate by defining a convenience function in `src/main/routing/packet.h`:

```C
/*                                                                                                                                                                                                                                                                                                                                                                                                              
 * Returns the packet's total length in bytes, including both header and payload fields.                                                                                                                                                                                                                                                                                                                        
 */                                                                                                                                                                                                                                                                                                                                                                                                             
static inline gsize packet_getTotalLength(const Packet *packet) {                                                                                                                                                                                                                                                                                                                                               
    return (gsize) packet_getPayloadLength(packet) + (gsize) packet_getHeaderSize(packet);                                                                                                                                                                                                                                                                                                                      
}   
```

Reply:
OK I refactored a `packet_getTotalSize` function as suggested.